### PR TITLE
jwt authentication

### PR DESCRIPTION
**Pull request by junwei26**

### CODE REVIEW

File: backend/src/authentication/auth.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const jwt = require("jsonwebtoken");
+const tokenSecret = "test";
+
+module.exports.authenticateToken = (req, res, next) => {
+  const authHeader = req.headers["authorization"];
+  const token = authHeader && authHeader.split(" ")[1];
+
+  if (token == null) return res.sendStatus(401);
+  jwt.verify(token, tokenSecret, { maxAge: 86400 }, (err, user) => {
+    console.log(err);
```
Comment (rohit0718) [inline backend/src/authentication/auth.js:10]: can remove the console logs if not required.

File: backend/src/authentication/auth.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const jwt = require("jsonwebtoken");
+const tokenSecret = "test";
+
+module.exports.authenticateToken = (req, res, next) => {
+  const authHeader = req.headers["authorization"];
+  const token = authHeader && authHeader.split(" ")[1];
+
+  if (token == null) return res.sendStatus(401);
+  jwt.verify(token, tokenSecret, { maxAge: 86400 }, (err, user) => {
```
Comment (rohit0718) [inline backend/src/authentication/auth.js:9]: the 86400 can be declared as a const too

File: frontend/src/components/navBar/NavBarButtons.js [hunk 3]
```diff
@@ -19,7 +17,14 @@ const NavBarButtons = () => {
           <Nav.Link as={Link} to="/settings">
             Settings
           </Nav.Link>
-          <Nav.Link as={Link} to="/" onClick={() => setLogin(false)}>
+          <Nav.Link
+            as={Link}
+            to="/"
+            onClick={() => {
+              localStorage.setItem("currentUser", JSON.stringify({ isLoggedIn: false, token: "" }));
+              window.location.reload(false);
```
Comment (rohit0718) [inline frontend/src/components/navBar/NavBarButtons.js:25]: `false` is default param so can remove it

File: frontend/src/pages/login/index.js [hunk 4]
```diff
@@ -15,13 +14,10 @@ const authenticate = (email, password) => {
   } else if (password.length <= 0) {
     alert("Password cannot be empty!");
   } else {
-    return true;
+    localStorage.setItem("currentUser", JSON.stringify({ isLoggedIn: true, token: "" }));
```
Comment (rohit0718) [inline frontend/src/pages/login/index.js:17]: still pending the check against the db for the user and pass. also will need to generate a jwt in the backend to pass to the frontend. right now as long as email and pass are valid it authorizes user

File: frontend/src/pages/login/index.js [hunk 5]
```diff
@@ -54,10 +50,9 @@ const Login = (props) => {
           size="lg"
           type="submit"
           onClick={() => {
-            if (authenticate(email, password)) {
-              setLogin(true);
-              props.history.push("/");
-            }
+            authenticate(email, password);
```
Comment (rohit0718) [inline frontend/src/pages/login/index.js:53]: any reason u took out the if block here? same goes for taking out the true and false returns from the authenticate function. if the user and pass are not valid the page refreshes. is that intended?

File: frontend/src/pages/register/index.js [hunk 6]
```diff
@@ -67,10 +64,8 @@ const Register = (props) => {
           size="lg"
           type="submit"
           onClick={() => {
-            if (authenticate(email, password, confirmPassword)) {
```
Comment (rohit0718) [inline frontend/src/pages/register/index.js:70]: same comment as for the login page.

File: backend/src/authentication/auth.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const jwt = require("jsonwebtoken");
+const tokenSecret = "test";
+
+module.exports.authenticateToken = (req, res, next) => {
```
Comment (rohit0718) [inline backend/src/authentication/auth.js:4]: do u need to create routes for this and generateToken below? or are you going to use these functions in the login/register methods instead?

**Review by rohit0718 — CHANGES_REQUESTED** [review]

File: backend/src/authentication/auth.js (hunk 7 above)
Comment (junwei26) [reply backend/src/authentication/auth.js:4]: No need to create routes, can just use this as a middleware func for any route(any get or post request) that requires authentication

**Pull request closed without merging**